### PR TITLE
Issue#48 fgsm attack notebook

### LINKED ISSUE #48: Ноутбук с fgsm_attack
- Сделать ноутбук, где на вход инференса приходит картинка с шумами (с атакой)
- Получить оценку качества работы модели с атаками (можно выбрать несколько epsilon для исследования)
- Используем тестовую выборку

### PR DESCRIPTION
**Pull request by MarinaKupetskova**

### CODE REVIEW

**Review by serykhelena — CHANGES_REQUESTED** [review]
1. Здесь избыточен второй Compose
```
transform = transforms.Compose(
    [Invertor(),transforms.Compose([transforms.ToTensor()])]
    )
```
Можно проще 

```
transform = transforms.Compose(
    [
        Invertor(),
        transforms.ToTensor()
    ]
)
```

2  В функции data_fgsm_attack строка 35

```
perturbed_data = np.reshape(perturbed_data.detach().numpy(),(28,28))
```

Должно быть 

```
perturbed_data = np.reshape(perturbed_data.cpu().detach().numpy(),(28,28))
```

Иначе на машине с видяхой - падает 

3 Вынести в отдельную переменную SAVE_DPATH 

```
SAVE_DPATH = os.path.join(PROJECT_ROOT,'results','mnist_fgsm_attacks')
os.makedirs(SAVE_DPATH, exist_ok=True)
```
Когда папки mnist_fgsm_attacks нет, то код падает при сохранении 

4. Добавить в ноутбук визуализацию картинок. Исходная картинка - та же картинка с атакой (с указанием значения эпсилон). Можно через interact

**Review by serykhelena — CHANGES_REQUESTED** [review]
Было бы здорово добавить предсказание для исходной картинки (без атак) и предсказание с атаками, чтобы сразу увидеть влияние. А то сейчас не очень понятно Predicted Label к чему относится, т.к. 2 картинки идут потом  
image

**Comment by MarinaKupetskova** [comment]
Вывод метрик для оригинальных изображений оставить или лучше убрать?

**Review by serykhelena — APPROVED** [review]

**Comment by serykhelena** [comment]
> Вывод метрик для оригинальных изображений оставить или лучше убрать?

Пусть пока будут (=
